sdk/src/main/java/com/oursky/authgear/WebKitWebViewActivity.kt

```diff
@@ -0,0 +1,263 @@
+            }
+        }
+
+        fun toBundle(): Bundle {
+            val bundle = Bundle()
+            bundle.putParcelable("url", this.url)
+            bundle.putParcelable("redirectURI", this.redirectURI)
+            this.actionBarBackgroundColor?.let {
+                bundle.putInt("actionBarBackgroundColor", it)
+            }
+            this.actionBarButtonTintColor?.let {
+                bundle.putInt("actionBarButtonTintColor", it)
+            }
+            return bundle
+        }
+    }
+
+    private class MyWebViewClient constructor(private val activity: WebKitWebViewActivity) :
+        WebViewClient() {
+
+        @TargetApi(Build.VERSION_CODES.N)
+        override fun shouldOverrideUrlLoading(
+            view: WebView?,
+            request: WebResourceRequest?
+        ): Boolean {
+            val uri = request?.url!!
+            if (this.shouldOverrideUrlLoading(uri)) {
+                return true
+            }
+            return super.shouldOverrideUrlLoading(view, request)
+        }
+
+        @SuppressWarnings("deprecation")
+        override fun shouldOverrideUrlLoading(view: WebView?, url: String?): Boolean {
+            val uri = Uri.parse(url!!)!!
+            if (this.shouldOverrideUrlLoading(uri)) {
+                return true
+            }
+            return super.shouldOverrideUrlLoading(view, url)
+        }
+
+        private fun shouldOverrideUrlLoading(uri: Uri): Boolean {
+            if (this.checkRedirectURI(uri)) {
+                return true;
+            }
+            return false;
+        }
+
+        private fun checkRedirectURI(uri: Uri): Boolean {
+            val redirectURI = this.activity.getOptions().redirectURI
+            val withoutQuery = this.removeQueryAndFragment(uri)
+            if (withoutQuery.toString() ==  redirectURI.toString()) {
+                this.activity.result = uri
+                this.activity.callSetResult()
+                this.activity.finish()
+                return true
+            }
+            return false;
+        }
+
+        private fun removeQueryAndFragment(uri: Uri): Uri {
+            return uri.buildUpon().query(null).fragment(null).build()
+        }
+    }
+
+    private class MyWebChromeClient constructor(private val activity: WebKitWebViewActivity) :
+        WebChromeClient() {
+
+        @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
+        override fun onShowFileChooser(
+            webView: WebView?,
+            filePathCallback: ValueCallback<Array<Uri>>?,
+            fileChooserParams: FileChooserParams?
+        ): Boolean {
+            val handle = StartActivityHandle(TAG_FILE_CHOOSER, filePathCallback!!)
+            val requestCode = this.activity.handles.push(handle)
+            val intent = fileChooserParams!!.createIntent()
+            this.activity.startActivityForResult(intent, requestCode)
+            return true
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
```

Comment:
I think you might need to save the webview state in onSaveInstanceState, and restore the webview state later:

https://github.com/authgear/authgear-sdk-android/blob/2023C01-24112023/sdk/src/main/java/com/oursky/authgear/latte/fragment/LatteFragment.kt#L278C24-L278C36

Reply:
Seems we need to back port this fix to all SDKs too